components/AssetFormBuy/index.tsx

```diff
@@ -0,0 +1,105 @@
+import { ChangeEvent, Dispatch, SetStateAction } from 'react';
+import Button from '../Button';
+import { useAuthContext } from '../Provider';
+import {
+  ErrorMessage,
+  DropdownMenu,
+  General,
+  Amount,
+  Row,
+} from './AssetFormBuy.styled';
+import { SaleAsset } from '../../services/sales';
+
+type Props = {
+  dropdownAssets: SaleAsset[];
+  lowestPrice: string;
+  maxSupply: string;
+  buttonText: string;
+  saleId: string;
+  purchasingError: string;
+  isLoadingPrices: boolean;
+  formattedPricesBySaleId: {
+    [templateMint: string]: string;
+  };
+  handleButtonClick: () => void;
+  setPurchasingError: Dispatch<SetStateAction<string>>;
+  setIsBalanceInsufficient: Dispatch<SetStateAction<boolean>>;
+  setSaleId: Dispatch<SetStateAction<string>>;
+};
+
+export const AssetFormBuy = ({
+  dropdownAssets,
+  lowestPrice,
+  maxSupply,
+  buttonText,
+  saleId,
+  purchasingError,
+  isLoadingPrices,
+  formattedPricesBySaleId,
+  handleButtonClick,
+  setPurchasingError,
+  setIsBalanceInsufficient,
+  setSaleId,
+}: Props): JSX.Element => {
+  const { currentUser, currentUserBalance } = useAuthContext();
+  const balanceAmount = parseFloat(
+    currentUserBalance.split(' ')[0].replace(',', '')
+  );
+
+  const handleDropdownSelect = (e: ChangeEvent<HTMLSelectElement>) => {
+    setPurchasingError('');
+    setIsBalanceInsufficient(false);
+
+    if (!currentUser) {
+      setPurchasingError('You must log in to purchase an asset.');
+      return;
```

Comment:
Moving this comment over:
> I think it'll be better to not return out of the function so early here and in the insufficient balance area. We should move up the setSaleId and then show error so that it doesn't feel "broken".